src/zenml/zen_stores/rest_zen_store.py

```diff
@@ -3183,6 +3183,8 @@ def session(self) -> requests.Session:
                 )
 
             self._session = requests.Session()
+            self._session.adapters["http://"].max_retries = 5
+            self._session.adapters["https://"].max_retries = 5
```

Comment:
Are you sure this is ok ? From what I can tell based on the Adapter constructor, this field needs to be set to a `urllib.util.retry.Retry` object instead of an integer:

```
    def __init__(
        self,
        pool_connections=DEFAULT_POOLSIZE,
        pool_maxsize=DEFAULT_POOLSIZE,
        max_retries=DEFAULT_RETRIES,
        pool_block=DEFAULT_POOLBLOCK,
    ):
        if max_retries == DEFAULT_RETRIES:
            self.max_retries = Retry(0, read=False)
        else:
            self.max_retries = Retry.from_int(max_retries)
```

The second question is what exactly is being retried ? Does an HTTP error like 404 Not Found or 403 Forbidden also trigger the client to retry the request ? All HTTP errors are treated separately by ZenML code, and therefore this retry mechanism should only be triggered by TCP errors, such as a connection being closed unexpectedly or timing out.

Reply:
I implemented something similar [here](https://github.com/zenml-io/zenml/blob/0807bf62bf0d2af4171917fe234a65ef14d21c5d/src/zenml/zen_server/rbac/zenml_cloud_rbac.py#L257-L258). The `Retry` object also allows you to specify in which cases it retries.